### PR TITLE
Custom markup via Options parameter

### PR DESCRIPTION
**Pull request by hansds**
I was glad I could use this package in a project of mine, but needed functionality where two divs surrounded the line text. The effect I was aiming for can be seen in the title text here: https://www.beangels.eu/

I added a flexible solution where you could inject the necessary markup, see README.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { tokenizeEl, getLineElements, getLines } from './utils';
+import { tokenizeEl, getLineElements, getLines, Options } from './utils';
 
 const waitForBrowser = async () => {
   return new Promise(resolve => window.setTimeout(resolve, 0));
 };
 
-export default async (el: HTMLElement) => {
+export default async (el: HTMLElement, options?: Options | undefined) => {
```
Comment (djkramnik) [inline src/index.ts]: I don't believe you need the ` | undefined` here, is the use of ? sufficient?

File: src/index.ts [hunk 2]
```diff
@@ -15,7 +15,8 @@ export default async (el: HTMLElement) => {
   el.innerText = '';
   el.appendChild(tokens);
   await waitForBrowser();
-  const lines = getLineElements(getLines(el));
+
+  const lines = getLineElements(getLines(el), options?.markup);
```
Comment (djkramnik) [inline src/index.ts]:
```suggestion
  const lines = getLineElements(getLines(el), options ? options.markup : undefined);
```

File: src/utils.ts [hunk 3]
```diff
@@ -35,17 +39,28 @@ export const getLines = (tokenizedEl: HTMLElement) =>
     {}
   );
 
-export const getLineElements = (linesMap: LinesMapType) => {
+export const getLineElements = (
+  linesMap: LinesMapType,
+  markup: string | undefined
```
Comment (djkramnik) [inline src/utils.ts]:
```suggestion
  markup?: string
```

File: src/utils.ts [hunk 4]
```diff
@@ -35,17 +39,28 @@ export const getLines = (tokenizedEl: HTMLElement) =>
...
   Object.keys(linesMap)
     .sort((a: string, b: string) => parseInt(a, 10) - parseInt(b, 10))
     .forEach((key: string) => {
-      const div = document.createElement('div');
-      div.style.display = 'block';
-      div.style.textAlign = 'start';
-      div.style.position = 'relative';
-      div.textContent = linesMap[key].join(' ');
-      fragment.appendChild(div);
+      if (markup) {
```
Comment (djkramnik) [inline src/utils.ts]:
```suggestion
      if (typeof markup === 'string') {
```

**Review by djkramnik — CHANGES_REQUESTED** [review]
Thanks for your interest.  I made a few suggestions, but haven't explicitly tested them.  Also could you add something to the demo file that includes the added option.

File: src/index.ts (hunk 1 above)
Comment (hansds) [reply src/index.ts]: @djkramnik I added your suggestions, tested them and approved. I'm quite unexperienced in typescript I'm afraid!

**Comment by hansds** [comment]
@djkramnik thanks for your suggestions! I added them, and included a demo for the extended usage.

**Comment by hansds** [comment]
@djkramnik thanks for the accepted PR! Can you publish to npm so I can remove my fork and update my dependencies? Thanks!

**Pull request merged**